File: C#/CursoMicrosoftC#/CursoSourceProfessor/Curso2541/Modulo03/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient;
using System.Drawing;
using System.Text;
using System.Configuration;
using System.Windows.Forms;
using System.Transactions;

namespace Modulo03
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private SqlConnection ObjSqlConnection;
        private ConnectionStringSettings ObjConnectionStringSettings;
        private SqlTransaction ObjSqlTransaction;

        private void Cmd_TransacaoLocal_Click(object sender, EventArgs e)
        {
            ObjConnectionStringSettings = ConfigurationManager.ConnectionStrings["Curso2541"];
            ObjSqlConnection = new SqlConnection(ObjConnectionStringSettings.ConnectionString);

            try
            {
                ObjSqlConnection.Open();
                ObjSqlTransaction = ObjSqlConnection.BeginTransaction();

                SqlCommand CmDebito = new SqlCommand();
                CmDebito.Connection = ObjSqlConnection;
                CmDebito.CommandType = CommandType.StoredProcedure;
                CmDebito.CommandText = "P_DebitaContaCorrente";
                CmDebito.Transaction = ObjSqlTransaction;

                CmDebito.Parameters.Add(new SqlParameter("@Conta",SqlDbType.Int));
                CmDebito.Parameters.Add(new SqlParameter("@Valor",SqlDbType.Money));

                CmDebito.Parameters["@Conta"].Value = 2;
                CmDebito.Parameters["@Valor"].Value = 10M;

                CmDebito.ExecuteNonQuery();

                SqlCommand CmCredito = new SqlCommand();
                CmCredito.Connection = ObjSqlConnection;
                CmCredito.CommandType = CommandType.StoredProcedure;
                CmCredito.CommandText = "P_CreditoPoupanca";
                CmCredito.Transaction = ObjSqlTransaction;

                CmCredito.Parameters.Add(new SqlParameter("@Conta", SqlDbType.Int));
                CmCredito.Parameters.Add(new SqlParameter("@Valor", SqlDbType.Money));

                CmCredito.Parameters["@Conta"].Value = 1;
                CmCredito.Parameters["@Valor"].Value = 10M;

                CmCredito.ExecuteNonQuery();

                ObjSqlTransaction.Commit();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());

                ObjSqlTransaction.Rollback();
            }
        }

        private void Cmd_TransacaoDistribuida_Click(object sender, EventArgs e)
        {
            ConnectionStringSettings ObjStringCnCurso2541 = ConfigurationManager.ConnectionStrings["Curso2541"];
            ConnectionStringSettings ObjStringCnCurso2541B = ConfigurationManager.ConnectionStrings["Curso2541B"];
            
            SqlConnection ObjCnCurso2541 = new SqlConnection(ObjStringCnCurso2541.ConnectionString);
            SqlConnection ObjCnCurso2541B = new SqlConnection(ObjStringCnCurso2541B.ConnectionString);

            using (TransactionScope ObjTransactionScope = new TransactionScope())
            {
                try
                {
                    ObjCnCurso2541.Open();

                    SqlCommand CmDebito = new SqlCommand();
                    CmDebito.Connection = ObjCnCurso2541;
                    CmDebito.CommandType = CommandType.StoredProcedure;
                    CmDebito.CommandText = "P_DebitaContaCorrente";

                    CmDebito.Parameters.Add(new SqlParameter("@Conta", SqlDbType.Int));
                    CmDebito.Parameters.Add(new SqlParameter("@Valor", SqlDbType.Money));

                    CmDebito.Parameters["@Conta"].Value = 2;
                    CmDebito.Parameters["@Valor"].Value = 10M;

                    CmDebito.ExecuteNonQuery();

                    ObjCnCurso2541B.Open();

                    SqlCommand CmCredito = new SqlCommand();
                    CmCredito.Connection = ObjCnCurso2541B;
                    CmCredito.CommandType = CommandType.StoredProcedure;
                    CmCredito.CommandText = "P_CreditoPoupanca";

                    CmCredito.Parameters.Add(new SqlParameter("@Conta", SqlDbType.Int));
                    CmCredito.Parameters.Add(new SqlParameter("@Valor", SqlDbType.Money));

                    CmCredito.Parameters["@Conta"].Value = 1;
                    CmCredito.Parameters["@Valor"].Value = 10M;

                    CmCredito.ExecuteNonQuery();

                    ObjTransactionScope.Complete();
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.ToString());
                }
                finally
                {
                    if (ObjCnCurso2541.State == ConnectionState.Open)
                    {
                        ObjCnCurso2541.Close();
                    }
                    if (ObjCnCurso2541B.State == ConnectionState.Open)
                    {
                        ObjCnCurso2541B.Close();
                    }
                }
            }
        }
    }
}

File: C#/CursoMicrosoftC#/CursoSourceProfessor/Curso2541VB/Modulo03/Form1.vb
Imports System.Data
Imports System.Data.SqlClient
Imports System.Transactions
Imports System.Configuration

Public Class Form1
    Private ObjSqlConnection As SqlConnection
    Private ObjConnectionStringSettings As ConnectionStringSettings
    Private ObjSqlTransaction As SqlTransaction

    Private Sub Cmd_TransacaoLocal_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cmd_TransacaoLocal.Click
        ObjConnectionStringSettings = ConfigurationManager.ConnectionStrings("Curso2541")
        ObjSqlConnection = New SqlConnection(ObjConnectionStringSettings.ConnectionString)

        Try
            ObjSqlConnection.Open()
            ObjSqlTransaction = ObjSqlConnection.BeginTransaction()

            Dim CmDebito As New SqlCommand()
            CmDebito.Connection = ObjSqlConnection
            CmDebito.CommandType = CommandType.StoredProcedure
            CmDebito.CommandText = "P_DebitaContaCorrente"
            CmDebito.Transaction = ObjSqlTransaction

            CmDebito.Parameters.Add(New SqlParameter("@Conta", SqlDbType.Int))
            CmDebito.Parameters.Add(New SqlParameter("@Valor", SqlDbType.Money))

            CmDebito.Parameters("@Conta").Value = 2
            CmDebito.Parameters("@Valor").Value = 10

            CmDebito.ExecuteNonQuery()

            Dim CmCredito As New SqlCommand()
            CmCredito.Connection = ObjSqlConnection
            CmCredito.CommandType = CommandType.StoredProcedure
            CmCredito.CommandText = "P_CreditoPoupanca"
            CmCredito.Transaction = ObjSqlTransaction

            CmCredito.Parameters.Add(New SqlParameter("@Conta", SqlDbType.Int))
            CmCredito.Parameters.Add(New SqlParameter("@Valor", SqlDbType.Money))

            CmCredito.Parameters("@Conta").Value = 1
            CmCredito.Parameters("@Valor").Value = 10

            CmCredito.ExecuteNonQuery()

            ObjSqlTransaction.Commit()
        Catch ex As Exception

            MessageBox.Show(ex.ToString())

            ObjSqlTransaction.Rollback()
        End Try

    End Sub

    Private Sub Cmd_TransacaoDistribuida_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cmd_TransacaoDistribuida.Click
        Dim ObjStringCnCurso2541 As ConnectionStringSettings = ConfigurationManager.ConnectionStrings("Curso2541")
        Dim ObjStringCnCurso2541B As ConnectionStringSettings = ConfigurationManager.ConnectionStrings("Curso2541B")

        Dim ObjCnCurso2541 As New SqlConnection(ObjStringCnCurso2541.ConnectionString)
        Dim ObjCnCurso2541B As New SqlConnection(ObjStringCnCurso2541B.ConnectionString)

        Using ObjTransactionScope = New TransactionScope()
            Try
                ObjCnCurso2541.Open()

                Dim CmDebito As New SqlCommand()
                CmDebito.Connection = ObjCnCurso2541
                CmDebito.CommandType = CommandType.StoredProcedure
                CmDebito.CommandText = "P_DebitaContaCorrente"

                CmDebito.Parameters.Add(New SqlParameter("@Conta", SqlDbType.Int))
                CmDebito.Parameters.Add(New SqlParameter("@Valor", SqlDbType.Money))

                CmDebito.Parameters("@Conta").Value = 2
                CmDebito.Parameters("@Valor").Value = 10

                CmDebito.ExecuteNonQuery()

                ObjCnCurso2541B.Open()

                Dim CmCredito As New SqlCommand()
                CmCredito.Connection = ObjCnCurso2541B
                CmCredito.CommandType = CommandType.StoredProcedure
                CmCredito.CommandText = "P_CreditoPoupanca"

                CmCredito.Parameters.Add(New SqlParameter("@Conta", SqlDbType.Int))
                CmCredito.Parameters.Add(New SqlParameter("@Valor", SqlDbType.Money))

                CmCredito.Parameters("@Conta").Value = 1
                CmCredito.Parameters("@Valor").Value = 10

                CmCredito.ExecuteNonQuery()

                ObjTransactionScope.Complete()
            Catch ex As Exception
                MessageBox.Show(ex.ToString())
            Finally
                If (ObjCnCurso2541.State = ConnectionState.Open) Then
                    ObjCnCurso2541.Close()
                End If
                If (ObjCnCurso2541B.State = ConnectionState.Open) Then

                    ObjCnCurso2541B.Close()
                End If
            End Try
        End Using
    End Sub
End Class
